echoregions/regions2d/regions2d.py

```diff
@@ -107,132 +110,128 @@ def select_region(
-                                             and 1st index value is {time_range[1]}"
-                            )
-                    else:
+            elif isinstance(region_id, list):
+                for value in region_id:
+                    if not isinstance(value, (float, int, str)):
                         raise TypeError(
-                            f"Invalid time_range value types: \
-                                        {type(time_range[0])} and {type(time_range[1])}. Must \
-                                        be both of type Timestamp."
+                            f"Invalid element in list region_id. Is of type: "
+                            f"{type(value)}. Must be of type float, int, str."
                         )
-                else:
-                    raise ValueError(
-                        f"Invalid time_range size: {len(time_range)}. \
-                        Must be of size 2."
-                    )
             else:
                 raise TypeError(
-                    f"Invalid time_range type: {type(time_range)}. Must be \
-                                of type List."
+                    f"Invalid region_id type: {type(region_id)}. "
+                    "Must be of type float, int, str, list, or None."
                 )
-        if depth_range is not None:
-            if isinstance(depth_range, List):
-                if len(depth_range) == 2:
-                    if isinstance(depth_range[0], (float, int)) and isinstance(
-                        depth_range[1], (float, int)
-                    ):
-                        if depth_range[0] < depth_range[1]:
-                            # Select rows with depth values that are all within depth range
-                            region = region[
-                                region["time"].apply(
-                                    lambda depth_array: all(
-                                        depth_range[0] <= float(x) and depth_range[1] >= float(x)
-                                        for x in depth_array
-                                    )
-                                )
-                            ]
-                        else:
-                            raise ValueError(
-                                f"1st index value must be later than 0th index \
-                                             value. Currently 0th index value is {depth_range[0]} \
-                                             and 1st index value is {depth_range[1]}"
-                            )
-                    else:
+            region = region[region["region_id"].isin(region_id)]
+
+        # Check and subset for region_class
+        if region_class is not None:
+            if isinstance(region_class, str):
+                region_class = [region_class]
+            elif isinstance(region_class, list):
+                for value in region_class:
+                    if not isinstance(value, str):
```

Comment:
same as the `region_id` comment: seems like the list comprehension you have below for the timestamp part looks more succinct?
```
all(isinstance(t, Timestamp) for t in time_range)
```

Reply:
Same as the above, I'll resolve this now.